### PR TITLE
Demo fix

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Fixes two regressions related to demo display recently made in develop.

### CODE REVIEW

File: src/global/types.h [hunk 1]
```diff
@@ -1226,11 +1226,17 @@ typedef struct RGBA8888 {
     uint8_t a;
 } RGBA8888;
 
-typedef struct VECTOR_3D {
+typedef struct XYZ16 {
+    int16_t x;
+    int16_t y;
+    int16_t z;
+} XYZ16;
```
Comment (walkawayy) [inline src/global/types.h]: `XYZ_16` seems more readable IMO. Or maybe `Vector_16`? Same for 32 idk.

**Review by walkawayy — APPROVED** [review]
Just 1 comment. Otherwise demos and game seem to work fine.

File: src/global/types.h (hunk 1 above)
Comment (rr-) [reply src/global/types.h]: Is it OK for me then to convert RGB888 and the family to RGB_888? @walkawayy

Comment (walkawayy) [reply src/global/types.h]: That makes sense to me. If you agree.

**Pull request merged**